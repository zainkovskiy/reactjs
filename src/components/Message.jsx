import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class Message extends Component{
  static propTypes = {
    author: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  };
  render() {
    return(
        <div>
          { this.props.author }: { this.props.text }
        </div>
    )
  }

}