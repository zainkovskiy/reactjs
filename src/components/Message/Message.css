.message{
    max-width: 20%;
    border-radius: 15px;
    padding: 10px;
    hyphens: auto;
    position: relative;
}

.message-user{
    background-color: white;
    align-self: flex-start;
}
.message-user::after{
    content: "";
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 5px 18px 5px 0;
    border-color: transparent #fff transparent transparent;
    position: absolute;
    left: -10px;
    bottom: 2px;
}

.message-bot{
    background-color: #c6d9fd;
    align-self: flex-end;
}
.message-bot::after{
    content: "";
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 5px 0px 5px 18px;
    border-color: transparent transparent transparent #c6d9fd;
    position: absolute;
    right: -10px;
    bottom: 2px;
}

.message-text{
    font-size: 24px;
    font-family: sans-serif;
}

.message-author{
    color: #777777;
    font-size: 14px;
    font-family: sans-serif;
    font-weight: bold;
    margin-bottom: 10px;
}
