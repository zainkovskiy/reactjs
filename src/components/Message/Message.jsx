import './Message.css';

import React, { Component } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

export class Message extends Component{
  static propTypes = {
    author: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  };
  render() {
    const classes = classNames('message', {
      'message-user': this.props.author !== 'Bot',
      'message-bot': this.props.author === 'Bot',
    });
    return(
        <div className={classes}>
          <div className='message-author'>{ this.props.author }</div>
          <div className='message-text'>{ this.props.text }</div>
        </div>
    )
  }

}