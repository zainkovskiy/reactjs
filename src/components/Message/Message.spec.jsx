import React from "react";
import { Message } from "./Message";

import renderer from "react-test-renderer";
import { shallow } from "enzyme";

test("Message show message", () => {
  const tree = renderer
    .create(<Message author={"test1"} text={"test1"}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Message show div", () => {
  const header = shallow(<Message author={"test1"} text={"test1"}/>);
  const divs = header.find('div');

  expect(divs.length).toBe(3);
})