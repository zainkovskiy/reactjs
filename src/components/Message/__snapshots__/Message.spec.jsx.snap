// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message show message 1`] = `
<div
  className="message message-user"
>
  <div
    className="message-author"
  >
    test1
  </div>
  <div
    className="message-text"
  >
    test1
  </div>
</div>
`;
